python/lsst/pipe/tasks/hips.py

```diff
@@ -604,3 +614,907 @@ def build_quantum_graph(
+    png_gray_asinh_stretch = pexConfig.Field(
+        doc="AsinhMapping linear stretch for grayscale png scaling.",
+        dtype=float,
+        default=2.0,
+    )
+    png_gray_asinh_softening = pexConfig.Field(
+        doc="AsinhMapping softening parameter (Q) for grayscale png scaling.",
+        dtype=float,
+        default=8.0,
+    )
+
+
+class GenerateHipsTask(pipeBase.PipelineTask):
+    """Task for making a HiPS tree with fits and grayscale pngs."""
+    ConfigClass = GenerateHipsConfig
+    _DefaultName = "generateHips"
+
+    @timeMethod
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+
+        dims = inputRefs.hips_exposure_handles[0].dataId.names
+        order = None
+        for dim in dims:
+            if "healpix" in dim:
+                order = int(dim.split("healpix")[1])
+                healpix_dim = dim
+        if order is None:
+            raise RuntimeError("Could not determine healpix order for input exposures.")
+
+        hips_exposure_handle_dict = {
+            (hips_exposure_handle.dataId[healpix_dim],
+             hips_exposure_handle.dataId['band']): hips_exposure_handle
+            for hips_exposure_handle in inputs["hips_exposure_handles"]
+        }
+
+        band = butlerQC.quantum.dataId['band']
+
+        self.run(bands=[band], max_order=order, hips_exposure_handle_dict=hips_exposure_handle_dict)
+
+    def run(self, bands, max_order, hips_exposure_handle_dict, do_color=False):
+        """Run the GenerateHipsTask.
+
+        Parameters
+        ----------
+        bands : `list [ `str` ]
+            List of bands to be processed (or single band).
+        max_order : `int`
+            HEALPix order of the maximum (native) HPX exposures.
+        hips_exposure_handle_dict : `dict` {`int`: `lsst.daf.butler.DeferredDatasetHandle`}
+            Dict of handles for the HiPS high-resolution exposures.
+            Key is (pixel number, ``band``).
+        do_color : `bool`, optional
+            Do color pngs instead of per-band grayscale.
+        """
+        min_order = self.config.min_order
+
+        if not do_color:
+            png_grayscale_mapping = AsinhMapping(
+                self.config.png_gray_asinh_minimum,
+                self.config.png_gray_asinh_stretch,
+                Q=self.config.png_gray_asinh_softening,
+            )
+        else:
+            png_color_mapping = AsinhMapping(
+                self.config.png_color_asinh_minimum,
+                self.config.png_color_asinh_stretch,
+                Q=self.config.png_color_asinh_softening,
+            )
+
+            bcb = self.config.blue_channel_band
+            gcb = self.config.green_channel_band
+            rcb = self.config.red_channel_band
+            colorstr = f'{bcb}{gcb}{rcb}'
+
+        # The base path is the hips_base_uri with the band as a sub-directory.
+        hips_base_path = ResourcePath(self.config.hips_base_uri, forceDirectory=True)
```

Comment:
The comment says the base path should have the band included but then uses a variable called base_path that doesn't include the band.